// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe.autoflip;

import "mediapipe/examples/desktop/autoflip/quality/visual_scorer.proto";
import "mediapipe/framework/calculator.proto";

// Next tag: 6
message FaceToRegionCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional FaceToRegionCalculatorOptions ext = 282401234;
  }

  // Options for generating a score for the entire face from its visual
  // appearance. The generated score is used to modulate the detection scores
  // for whole face and/or landmark bbox region types.
  optional VisualScorerOptions scorer_options = 1;

  // If true, export the large face bounding box generated by the face tracker.
  // This bounding box is generally larger than the actual face and relatively
  // inaccurate.
  optional bool export_whole_face = 2 [default = false];

  // If true, export a number of individual face landmarks (eyes, nose, mouth,
  // ears etc) as separate SalientRegion protos.
  optional bool export_individual_face_landmarks = 3 [default = false];

  // If true, export two bounding boxes from landmarks (one for the core face
  // landmarks like eyes and nose, and one for extended landmarks including ears
  // and chin).
  optional bool export_bbox_from_landmarks = 4 [default = true];

  // If true, generate a score from the appearance of the face and use it to
  // modulate the detection scores for whole face and/or landmark bboxes.
  optional bool use_visual_scorer = 5 [default = true];
}
